service/history/historyBuilder.go

```diff
@@ -453,11 +453,11 @@ func (b *historyBuilder) addTransientEvent(event *historypb.HistoryEvent) *histo
 }
 
 func (b *historyBuilder) newWorkflowExecutionStartedEvent(
-	startRequest *historyservice.StartWorkflowExecutionRequest, previousExecution *persistence.WorkflowExecutionInfo, firstRunID, originalRunID string) *historypb.HistoryEvent {
+	startRequest *historyservice.StartWorkflowExecutionRequest, previousExecution *persistenceblobs.WorkflowExecutionInfo, previousExecutionState *persistenceblobs.WorkflowExecutionState, firstRunID, originalRunID string) *historypb.HistoryEvent {
 	var prevRunID string
 	var resetPoints *workflowpb.ResetPoints
-	if previousExecution != nil {
-		prevRunID = previousExecution.ExecutionState.RunId
+	if previousExecution != nil && previousExecutionState != nil {
+		prevRunID = previousExecutionState.RunId
 		resetPoints = previousExecution.AutoResetPoints
 	}
```

Comment:
will there be any case when `previousExecution != nil XOR previousExecutionState != nil` is true?

Reply:
I guess not. But because these are protos, I can remove this check all together.